topdown/http.go

```diff
@@ -647,11 +672,31 @@ func insertIntoHTTPSendInterQueryCache(bctx BuiltinContext, key ast.Value, resp
 
 	requestCache := bctx.InterQueryBuiltinCache
 
-	pcv, err := newInterQueryCacheValue(resp, respBody)
+	var obj ast.Object
+	var ok bool
+	if obj, ok = key.(ast.Object); !ok {
+		return fmt.Errorf("interface conversion error")
+	}
+
+	cachingMode, err := getCachingMode(obj)
 	if err != nil {
 		return err
 	}
 
+	var pcv cache.InterQueryCacheValue
+
+	if cachingMode == defaultCachingMode {
+		pcv, err = newInterQueryCacheValue(resp, respBody)
+		if err != nil {
+			return err
+		}
+	} else {
+		pcv, err = newInterQueryCacheData(resp, respBody)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
[nit]
```suggestion
	if cachingMode == defaultCachingMode {
		pcv, err = newInterQueryCacheValue(resp, respBody)
	} else {
		pcv, err = newInterQueryCacheData(resp, respBody)
	}
	if err != nil {
		return err
	}
```
but this is just a matter of taste I think 😄

Reply:
Updated